### ISSUE TITLE
Question to annotation from trusted proteins file

### ISSUE THREAD

**Issue by hezscha** [tseemann/prokka#411]
Hi, 

I'm running prokka with the --proteins option for annotation and I wanted to ask how it decides which matching annotation to report. The reason I'm asking is that I have observed the following case for my data: 

Prokka reports the following gff line for a predicted gene indicating that it is similar to PA0979:
NODE_1_length_515040_cov_21.358437	Prodigal:2.6	CDS	322802	323110	.	+	0	ID=ERR431416_00330;Parent=ERR431416_00330_gene;inference=ab initio prediction:Prodigal:2.6,similar to AA sequence:Pseudomonas_aeruginosa_PAO1_107.faa:PA0979;locus_tag=ERR431416_00330;note=ref|NP_249670.1|gi|15596176| conserved hypothetical protein;product=hypothetical protein

However when I manually blasted PAO1 genes against this sample before I found that gene PA1937 is actually the better match for this particular predicted gene:
BLAST output, fields are "qseqid qlen sseqid slen pident length mismatch gapopen qstart qend sstart send evalue"
PA1937	309	NODE_1_length_515040_cov_21.358437	515040	98.706	309	4	0	1	309	322802	323110	3.95e-154
PA0979	309	NODE_1_length_515040_cov_21.358437	515040	95.146	309	15	0	1	309	322802	323110	6.26e-139

So could I do something to make prokka report PA1937 instead of PA0979 as match?

Thanks,
Henrike Zschach

**Comment by tseemann** [comment]
What BLASTP parameters did you use?
If they aren't the same as Prokka then you will get different results.
The main one might be `-seg no`

```
blastp -query - -db %d -evalue %e -qcov_hsp_perc %c -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no
```

**Comment by hezscha** [comment]
Hi again, 

I was actually running blastn of the coding sequences of the trusted proteins against the assembly I ran prokka on. It turns out that PA0979 is closer to the predicted gene in protein space, as prokka also found, and PA1937 is closer in nucleotide space so perhaps PA0979 is the more appropriate tag. Thank you for the explanation.

**Comment by tseemann** [comment]
Is that possible? 
That a gene (CDS, DNA) is closer to X than Y (blastn N:N) 
but when using protein it is closer to Y than X (tblastn P:N) ?

**Comment by hezscha** [comment]
I just checked and PA0979 and PA1937 are also very close to each other, 3 amino acid substitutions apart.
 
I didn't do tblastn, but blastn on the CDS of the predicted gene against the CDS's of the two PAO1 genes as well as blastp on the predicted protein against the PAO1 proteins. The protein is 102 aa and the CDS 309 nucl. All alignments are full length. 
The predicted protein has 15 nucleotide substitutions to PA0979 (see blastn results in first post) that result in 2 compatible amino acid substitutions (blastp). To PA1937 there are 4 nucleotide substitutions that result in 3 amino acid substitutions. 

TLDR: Yes, it seems the CDS of the predicted gene is closer to X but the translation is marginally closer to Y. But they are also all very close to each other.

**Comment by tseemann** [comment]
Can you paste the full alignments from the standard BLAST report?
Are they a single HSP?

**Comment by tseemann** [comment]
PA0797 is the best protein:protein alignment score, so Prokka chooses it.  Prokka isn't an annotation transfer tool. I'm sorry, there is nothing you can do to change this. it would require implementing a `blastn` module for Prokka, which is in he TO-DO list but won't be appearing very soon.

Thank you for sharing the BLAST results. It is a good reminder that you can have more divergence in NUCL but less in PROT - it all depends where the SNPs occur!

**Comment by hezscha** [comment]
I completely concur with you and I also think now that PA0797 is the correct choice. I just made the mistake of assuming that a better nucleotide alignment would correspond to a better protein alignment which I can see now is not true. Thanks for looking into it!
